### PR TITLE
feat: add Edit Page button to insight pages

### LINKED ISSUE #685: Feature: Edit button on insight pages
### Type of feature

🍕 Feature

### Current behavior

There is not edit button on the dashboard

### Suggested solution

Insight Page - Repositories - Period v04

https://www.figma.com/file/OpVX6WT7dmWqnwRuEvADMF/OpenSauced-%E2%80%94-Insights-(insights.opensauced.pizza)?node-id=5176%3A6423&t=jZIlELgoFGnOMY8Y-0

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Deadreyo**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

Adds an Edit Button inside of existing Insight Pages to navigate easily to editing the page from inside.
## Related Tickets & Documents

fixes #685 
## Mobile & Desktop Screenshots/Recordings
firefox_NRyEBqMFbM
image

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — CHANGES_REQUESTED** [review]
Edit button should only appear for the owner (public pages show edit button) and logged in users

Screen Shot 2022-12-26 at 7 45 19 AM

**Review by bdougie — APPROVED** [review]
Looks good. Let's wait on a second review to confirm the path leveraging user ids from supabase is the ideal path.

File: components/molecules/InsightHeader/insight-header.tsx
Comment (NsdHSO) [inline components/molecules/InsightHeader/insight-header.tsx:41]: Wouldn't it be better that way? ``href={`${String(`${...}`)}`}``

**Review by NsdHSO — COMMENTED** [review]
LGTM

File: components/molecules/InsightHeader/insight-header.tsx
Comment (bdougie) [reply components/molecules/InsightHeader/insight-header.tsx:41]: No, this is not necessary.

@NsdHSO Please provide suggestions and avoid open-ended questions. What is the value? `String(...)`?

File: layouts/hub-page.tsx
Comment (brandonroberts) [inline layouts/hub-page.tsx:30]: Use `sub` here instead of `provider_id` from the API

```suggestion
            <InsightHeader insight={insight} repositories={repositories} insightId={insightId} isOwner={user?.user_metadata.sub === insight?.user_id} />
```

File: components/molecules/InsightHeader/insight-header.tsx
Comment (brandonroberts) [inline components/molecules/InsightHeader/insight-header.tsx:10]: nit: move third-party imports to the top of the file

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/molecules/InsightHeader/insight-header.tsx
Comment (Deadreyo) [reply components/molecules/InsightHeader/insight-header.tsx:10]: I assume you mean delete the empty lines between the imports right?

Comment (brandonroberts) [reply components/molecules/InsightHeader/insight-header.tsx:10]: Those are fine. Mainly just move things we import from packages to the top, followed by imports external to this folder in general, then local relative imports.

Comment (Deadreyo) [reply components/molecules/InsightHeader/insight-header.tsx]: like this?

**Review by brandonroberts — APPROVED** [review]

File: layouts/hub-page.tsx
Comment (0-vortex) [inline layouts/hub-page.tsx:19]: This is not checking for errors; checking for errors, the `insight?.user_id` can be simplified to `insight.user_id`

File: layouts/hub-page.tsx
Comment (0-vortex) [inline layouts/hub-page.tsx:30]: if we need the user id `user?.user_metadata.sub` in more places in the app, it could be a good idea to return it from the hook, right next to `user`

**Review by 0-vortex — CHANGES_REQUESTED** [review]
Couple of lines on line 32 IMHO, simple error and owner checking should ideally happen where the insight hook is first called; this can rotate the props sent to the header to provide errors (when any), and correctly differentiate between public, logged-in public, and logged-in owners.

Assume this is only adding the button, there should be some URL deduplication guards to check for the same things mentioned above and disable white/broken or public edit links. Is this accounted for in another issue?

File: layouts/hub-page.tsx
Comment (Deadreyo) [reply layouts/hub-page.tsx:30]: Wouldn't that be better done in another issue, in which we would see where user ids are needed in the app and change it?

Comment (0-vortex) [reply layouts/hub-page.tsx:30]: no, because it's not used anywhere else, that is also why I am suggesting doing it here; your PR would benefit most from it and it would help future contributors.

**Comment by Deadreyo** [comment]
I tried a trick from google to fix the showing of commits from previous PRs and it worked, but it seems ot have deleted some of the conversations that were related to them. Sorry for that @0-vortex

**Comment by 0-vortex** [comment]
> I tried a trick from google to fix the showing of commits from previous PRs and it worked, but it seems ot have deleted some of the conversations that were related to them. Sorry for that @0-vortex

Np, it's still in the previous commit history conversation, didn't disappear fully just got hidden! 🍕

File: lib/hooks/useSupabaseAuth.ts
Comment (NsdHSO) [inline lib/hooks/useSupabaseAuth.ts:10]:
```suggestion
  const [userId, setUserId] = useState<string | null | undefined>();
```

Comment (0-vortex) [reply lib/hooks/useSupabaseAuth.ts:10]: Good catch, the type hard coding there is also not needed when you don't force set it to undefined values later in the code

Comment (Deadreyo) [reply lib/hooks/useSupabaseAuth.ts:10]: You're definitely correct

File: layouts/hub-page.tsx
Comment (Deadreyo) [reply layouts/hub-page.tsx:19]: How should I check for errors? You mean like put the logic separately in a variable and pass it instead?
```ts
let isOwner = insight && (userId === insight.user_id)
```
?

**Comment by brandonroberts** [comment]
@Deadreyo deploy preview failed due to type error during build. Everything else LGTM

**Pull request merged**
